<a href="https://colab.research.google.com/github/Akshat123design/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Akshat123design/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:

# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


My lane: Search Query Classification. I want to study how search queries can be classified using available search data so that SEO teams can better decide which queries deserve attention. I chose this lane because it is practical, related to search data, and gives me a clear machine learning problem to explore during the internship.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


My work aims to improve the decision of which search queries should receive more attention from an SEO team. An SEO analyst or content team could use the classification to prioritize queries for further analysis and content planning. A wrong recommendation could cause the team to spend time on a low-value query or overlook a useful query, leading to inefficient use of time and resources.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/Akshat123design/flyrank-ml-internship-starter.git

fatal: destination path 'flyrank-ml-internship-starter' already exists and is not an empty directory.


In [ ]:
!find flyrank-ml-internship-starter -name "content_refresh_anonymized.csv"

flyrank-ml-internship-starter/data/raw/content_refresh_anonymized.csv


In [ ]:
import pandas as pd

df = pd.read_csv("flyrank-ml-internship-starter/data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Average 90-day impressions:", round(df["impressions_90d"].mean(), 2))
print("Average 90-day clicks:", round(df["clicks_90d"].mean(), 2))


Number of rows: 30000
Average 90-day impressions: 5200.37
Average 90-day clicks: 16.1


The starter dataset contains 30,000 rows. The average page has about 5,200.37 impressions over 90 days but only 16.1 clicks on average. This suggests that there is enough search activity to study which queries or pages may deserve more attention, while the relatively low average clicks make prioritization worth investigating.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will provide observed and measured patterns in the search data and can give directional, decision-support recommendations about which search queries or pages may deserve more attention. It will help prioritize opportunities based on the available data. It will not provide causal proof that one factor caused another, and it will not predict Google's ranking decisions or guarantee future search performance.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.